components/clp-py-utils/clp_py_utils/result.py

```diff
@@ -0,0 +1,10 @@
+from typing import Optional
+
+
+class Result:
```

Comment:
Can we use https://pypi.org/project/result/ instead? It has nicer semantics since it only allows a Result OR an Error, whereas the current code requires that the caller configures that manually.